beacon_node/beacon_chain/src/fetch_blobs/mod.rs

```diff
@@ -358,17 +358,24 @@ async fn compute_custody_columns_to_import<T: BeaconChainTypes>(
                 // `DataAvailabilityChecker` requires a strict match on custody columns count to
                 // consider a block available.
                 let mut custody_columns = data_columns_result
-                    .map(|mut data_columns| {
-                        data_columns.retain(|col| custody_columns_indices.contains(&col.index));
+                    .map(|data_columns| {
                         data_columns
+                            .into_iter()
+                            .filter(|col| custody_columns_indices.contains(&col.index))
+                            .map(|col| {
+                                KzgVerifiedCustodyDataColumn::from_asserted_custody(
+                                    KzgVerifiedDataColumn::from_execution_verified(col),
```

Comment:
We haven't removed the gossip verification in fetch_blobs_v1 above which does kzg verification. Do we want to remove that as well?

Reply:
Yeah I didn't do it because i thought the benefit was minimal with the blob count, and it requires introducing code changes to check gossip cache and da checker. But maybe it's worth doing it to keep things consistent so it's less confusing when we look at it in the future.

Reply:
i'm good with that.

Reply:
Thanks! I don't have a strong preference either will keep it as it is and wait for @michaelsproul's review.

Reply:
@jimmygchen What are the changes to check the gossip cache and DA checker required in the blobs/engine_v1 case? Isn't it similar in that we still do those checks prior to publishing? Here:

https://github.com/sigp/lighthouse/blob/cfb1f7331064b758c6786e4e1dc15507af5ff5d1/beacon_node/network/src/network_beacon_processor/mod.rs#L975

But yeah I'm happy to leave the blobs case as-is. No need to complicate it if the current logic works. Or we could clean it up and make it consistent, but in a separate PR?